tests/nested/prepare-image-check/task.yaml

```diff
@@ -0,0 +1,12 @@
+summary: check that prepare-image worked well from inside the generated image
+systems: [ubuntu-core-16-*]
+manual: true
+execute: |
+    echo "Wait for first boot to be done"
+    while ! snap changes | grep -q "Done.*Initialize system state"; do sleep 1; done
+
+    echo "We have a model assertion"
+    snap known model|grep "series: 16"
+
+    echo "Make sure core has an actual revision"
+    snap list | MATCH "^core +[0-9]+\.[0-9]+ +[0-9]+ +canonical +\-"
```

Comment:
One extra test that would be nice is to check if `snap info |MATCH "tracking: (stable|candidate|beta|edge)"` (not sure if MATCH supports this syntax. I.e. ensure we are tracking a channel.